lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -2913,6 +2915,341 @@ def permutation_matrices(n):
+        if 'zorder' not in kwargs:
+            kwargs['zorder'] = 2
+        plot_line_style = {
+            **base_style,
+            **kwargs,
+            'zorder': (kwargs['zorder'] - .1 if barsabove else
+                       kwargs['zorder'] + .1),
+        }
+
+        # make the style dict for the line collections (the bars)
+        eb_lines_style = dict(base_style)
+        eb_lines_style.pop('marker', None)
+        eb_lines_style.pop('markerfacecolor', None)
+        eb_lines_style.pop('markeredgewidth', None)
+        eb_lines_style.pop('markeredgecolor', None)
+        eb_lines_style.pop('linestyle', None)
+        eb_lines_style['color'] = ecolor
+
+        if elinewidth:
+            eb_lines_style['linewidth'] = elinewidth
+        elif 'linewidth' in kwargs:
+            eb_lines_style['linewidth'] = kwargs['linewidth']
+
+        for key in ('transform', 'alpha', 'zorder', 'rasterized'):
+            if key in kwargs:
+                eb_lines_style[key] = kwargs[key]
+
+        # make the style dict for cap collections (the "hats")
+        eb_cap_style = dict(base_style)
+        # eject any marker information from format string
+        eb_cap_style.pop('marker', None)
+        eb_cap_style.pop('ls', None)
+        eb_cap_style['linestyle'] = 'none'
+        if capsize is None:
+            capsize = kwargs.pop('capsize', rcParams["errorbar.capsize"])
+        if capsize > 0:
+            eb_cap_style['markersize'] = 2. * capsize
+        if capthick is not None:
+            eb_cap_style['markeredgewidth'] = capthick
+        eb_cap_style['color'] = ecolor
+
+        if plot_line:
+            data_line = art3d.Line3D(x, y, z, **plot_line_style)
+            self.add_line(data_line)
+
+        try:
+            offset, errorevery = errorevery
+        except TypeError:
+            offset = 0
+
+        if errorevery < 1 or int(errorevery) != errorevery:
+            raise ValueError(
+                'errorevery must be positive integer or tuple of integers')
+        if int(offset) != offset:
+            raise ValueError("errorevery's starting index must be an integer")
+
+        everymask = np.zeros(len(x), bool)
+        everymask[offset::errorevery] = True
+
+        def _apply_mask(arrays, mask):
+            # Return, for each array in *arrays*, the elements for which *mask*
+            # is True, without using fancy indexing.
+            return [[*compress(array, mask)] for array in arrays]
+
+        def _extract_errs(err, data, lomask, himask):
+            # For separate +/- error values we need to unpack err
+            if len(err.shape) == 2:
+                low_err, high_err = err
+            else:
+                low_err, high_err = err, err
+
+            # for compatibility with the 2d errorbar function, when both upper
+            # and lower limits specified, we need to draw the markers / line -
+            # whether or not using both limits makes any sense (it doesn't)
```

Comment:
What doesn't make sense exactly?

Reply:
Upper limits and lower limits should technically be mutually exclusive on a data point - the true value of it is either below the arrow (upper limit) or above it (lower limit). It doesn't make much sense for both to be specified at the same time.

However, the 2d errorbar API allows for it - so the code block here replicates that functionality.

Reply:
Do you mean from a physical standpoint? I'm not sure I understand why it doesn't make sense from the library point of view.

Reply:
Maybe I felt grumpy when adding the comment - I'll remove that last bit.